stock_production_lot_warranty/models/stock_production_lot.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
+from odoo import api, fields, models
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+
+
+class StockProductionLot(models.Model):
+    _inherit = 'stock.production.lot'
+
+    warranty_exp_date = fields.Date(string='Warranty Expiration Date')
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        self.warranty_exp_date = False
+        if self.product_id and \
+                self.product_id.product_tmpl_id.warranty_type and \
+                self.product_id.product_tmpl_id.warranty:
+            if self.product_id.product_tmpl_id.warranty_type == 'day':
```

Comment:
@murtuzasaleh You could maybe simplify writing:
```suggestion
            warranty_type = self.product_id.product_tmpl_id.warranty_type
            if  warranty_type == 'day':
```
and so on

Reply:
@rousseldenis Thanks, Done.